src/metabase/models/values_card.clj

```diff
@@ -0,0 +1,83 @@
+(ns metabase.models.values-card
+  (:require
+   [clojure.string :as str]
+   [metabase.query-processor :as qp]
+   [metabase.util :as u]
+   [toucan.db :as db]
+   [toucan.models :as models]))
+
+
+(models/defmodel ValuesCard :values_card)
+
+(u/strict-extend #_{:clj-kondo/ignore [:metabase/disallow-class-or-type-on-model]} (class ValuesCard)
+                 models/IModel
+                 (merge models/IModelDefaults
+                        {:properties (constantly {:timestamped? true})
+                         :types      (constantly {:parameterized_object_type :keyword})}))
+
+(defn- search-match?
+  [search-term filter-term]
+  (str/includes? filter-term search-term))
+
+(defn query-matches
+  "Filter the values according to the `search-term`. If `search-term` is blank, return all the values without filtering"
+  [search-term values]
+  (if (str/blank? search-term)
+    values
+    (let [normalize              (comp str/lower-case str/trim)
+          normalized-search-term (normalize search-term)]
+      (filter #(search-match? normalized-search-term (normalize %)) values))))
+
+(defn- query-for-dashboard
+  [{dashboard-id :id} param-key]
+  (->> (db/query {:select [:dataset_query]
+                  :from [:report_card]
+                  :join [:values_card [:= :report_card.id :values_card.card_id]]
+                  :where [:and
+                          [:= :parameterized_object_id dashboard-id]
+                          [:= :parameterized_object_type "dashboard"]
+                          [:= :parameter_id param-key]]})
+       (db/do-post-select 'Card)
+       first
+       :dataset_query))
+
+(defn values-for-dashboard
+  "Returns a map with the filter `:values` associated with the dashboard."
+  [dashboard param-key search-term]
+  {:values
+   (->> (qp/process-query (query-for-dashboard dashboard param-key) {:rff (constantly conj)})
+        (map first)
```

Comment:
shouldn't we figure out how to get the correct column instead of defaulting to the 1st column?

Reply:
My understanding is that in Milestone 1 we assume the question has only a single column selected (or we unconditionally grab the first column). To support multiple columns would require input from a user about which one. That is a later milestone and we get a shippable state here with this assumption. Perhaps one thing we could do is when you choose a question we could assert (for the moment) that it only has a single column in it?

Reply:
if that's the same milestone as FE then I think it's okay. I think a good goal for merging is we have sth in master so that FE could start from there.

I'll check with AlexP about his plan.